api/src/org/apache/cloudstack/acl/Rule.java

```diff
@@ -0,0 +1,42 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.acl;
+
+import com.cloud.exception.InvalidParameterValueException;
+import com.google.common.base.Strings;
+
+public final class Rule {
+    private final String rule;
+    private final static String ALLOWED_PATTERN = "^[a-zA-Z0-9*]+$";
+
+    public Rule(final String rule) {
+        validate(rule);
+        this.rule = rule;
+    }
+
+    public String toString() {
+        return rule;
+    }
+
+    public static boolean validate(final String rule) throws InvalidParameterValueException {
+        if (Strings.isNullOrEmpty(rule) || !rule.matches(ALLOWED_PATTERN)) {
```

Comment:
Extract this validation logic into an immutable value class, ``Rule``, to encapsulate this validation logic and provide strong type specification in lower layers.